enable method for continuing results

jdlrobson opened the pull request:
when the bottom of the page is scrolled to it will retrieve the next
set of results
- for ajax results that go through monuments api that can be continued set next property on array
- add list item at end of the list to signal it is possible to continue
  *\* add dotted border bottom to suggest the list is unfinished
- when locating new monuments show ajax loader
- prevent multiple attempts to load new monuments by checking for an existing query

jrobson committed:
enable method for continuing results

when the bottom of the page is scrolled to it will retrieve the next
set of results

* for ajax results that go through monuments api that can be continued set next property on array
* add list item at end of the list to signal it is possible to continue
** add dotted border bottom to suggest the list is unfinished
* when locating new monuments show ajax loader
* prevent multiple attempts to load new monuments by checking for an existing query

bvibber commented:
I see the dotted line border on empty campaigns where there are no monuments...

I'm not so sure things are loading correctly either; picking Germany -> Nordrhein-Westfalen -> Koln -> Koln -> Altstadt-Nord ... when scrolling, the images that have already loaded disappear briefly, and sometimes things change in the sort order.

jdlrobson commented:
Rebased and fixed issue with empty campaigns - basically what was happening is there are monuments for Ghana but they are not being hidden as there is no information published on how to publish photos for that campaign (see bug https://bugzilla.wikimedia.org/show_bug.cgi?id=39109)

The issue with Germany -> Nordrhein-Westfalen -> Koln -> Koln -> Altstadt-Nord is an api issue..
Two things are happening
1) http://wlm.wikimedia.org/api/api.php?action=search&limit=50&uselang=en&sradm0=de&sradm1=de-nw&sradm2=K%C3%B6ln&sradm3=K%C3%B6ln&sradm4=Altstadt-Nord&format=json returns a json with url encoded srcontinue value ("ehem.%20Garage%20Gerling%20Konzern") - this SHOULD NOT BE ENCODED
2) When making a request to http://wlm.wikimedia.org/api/api.php?action=search&limit=50&uselang=en&sradm0=de&sradm1=de-nw&sradm2=K%C3%B6ln&sradm3=K%C3%B6ln&sradm4=Altstadt-Nord&format=json&srcontinue=Wohn-%2520und%2520Gesch%25C3%25A4ftshaus
instead of throwing a 400 bad request as it should it actually returns data where srcontinue value is "Wohn-%20und%20Gesch%C3%A4ftshaus" - so it ends up getting stuck in an infinite loop.

I'd rather not put yet more "gaffer tape" in the client to cope with this api problem.

jdlrobson commented:
Talking to Max this seems to be due to pagination being broken. Max is looking into this.

MaxSem commented:
Should be fixed now, please verify.

bvibber commented:
Merged manually.
